jdbl-core/src/main/java/se/kth/castor/jdbl/coverage/JacocoCoverage.java

```diff
@@ -0,0 +1,304 @@
+import org.jacoco.report.IReportVisitor;
+import org.jacoco.report.ISourceFileLocator;
+import org.jacoco.report.MultiReportVisitor;
+import org.jacoco.report.MultiSourceFileLocator;
+import org.jacoco.report.xml.XMLFormatter;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.ParserConfigurationException;
+import se.kth.castor.jdbl.debloat.DebloatTypeEnum;
+import se.kth.castor.jdbl.test.TestRunner;
+import se.kth.castor.jdbl.util.JarUtils;
+import se.kth.castor.jdbl.util.MavenUtils;
+
+/**
+ * This class runs JaCoCo via CLI.
+ *
+ * @see <a https://www.jacoco.org/jacoco/trunk/doc/cli.html</a>
+ */
+public class JacocoCoverage extends AbstractCoverage implements UsageAnalyzer
+{
+    private static final Logger LOGGER = LogManager.getLogger(JacocoCoverage.class.getName());
+
+    private Instrumenter instrumenter;
+    private File dest;
+    private List<File> source;
+    private List<File> execFiles;
+    private List<File> classFiles;
+    private List<File> sourceFiles;
+    private String name = "JaCoCo Coverage Report";
+    private File xml;
+    JacocoReportReader reportReader = null;
+
+    public JacocoCoverage(
+        MavenProject mavenProject,
+        File mavenHome,
+        DebloatTypeEnum debloatTypeEnum)
+    {
+        super(mavenProject, mavenHome, debloatTypeEnum);
+    }
+
+    @Override
+    public UsageAnalysis analyzeUsages()
+    {
+        try {
+            writeCoverage();
+        } catch (Exception e) {
+            LOGGER.error("Error writing coverage file.");
+        }
+
+        File report = new File(mavenProject.getBasedir().getAbsolutePath() + "/target/report.xml");
+
+        // Read the jacoco report
+        try {
+            reportReader = new JacocoReportReader();
+        } catch (ParserConfigurationException e) {
+            LOGGER.error("Error parsing jacoco.xml file.");
+        }
+        try {
+            assert reportReader != null;
+            return reportReader.getUsedClassesAndMethods(report);
+        } catch (IOException | SAXException e) {
+            LOGGER.error("Error getting unused classes and methods file.");
+        }
+
+        return null;
+    }
+
+    private void writeCoverage() throws IOException
+    {
+        LOGGER.info("Running JaCoCo");
+        final String baseDir = mavenProject.getBasedir().getAbsolutePath();
+        final String classesDir = baseDir + "/target/classes";
+        final String testDir = baseDir + "/target/test-classes";
+        final String instrumentedDir = baseDir + "/target/instrumented";
+        final String classesOriginalDir = baseDir + "/target/classes-original";
+
+        source = Arrays.asList(new File(classesDir));
+        dest = new File(instrumentedDir);
+        execFiles = Arrays.asList(new File(baseDir + "/jacoco.exec"));
+        classFiles = Arrays.asList(new File(classesDir));
+        sourceFiles = Arrays.asList(new File(mavenProject.getBasedir() + "/src/main/java"));
```

Comment:
The source code is not always in `/src/main/java`, this will introduce some regression compared to the current version of JDBL.

Reply:
Thanks for reviwing and reporting this issue, I just resolve it in https://github.com/castor-software/jdbl/commit/56643ee695469b15424a09883c8cb09b3818177d  by getting the list of `sourceFiles` directly from Maven:
```
sourceFiles = mavenProject
	.getCompileSourceRoots()
	.stream()
	.map(File::new)
	.collect(Collectors.toList());
```

BTW, the build-helper-maven-plugin is handy to add more src directories. See [here](https://www.baeldung.com/maven-project-multiple-src-directories).